lib/Driver/UnixToolChains.cpp

```diff
@@ -42,7 +42,8 @@ std::string
 toolchains::GenericUnix::sanitizerRuntimeLibName(StringRef Sanitizer,
                                                  bool shared) const {
   return (Twine("libclang_rt.") + Sanitizer + "-" +
-          this->getTriple().getArchName() + ".a")
+          this->getTriple().getArchName() +
+          (this->getTriple().isAndroid() ? "-android" : "") + ".a")
```

Comment:
[Needed similar to clang](https://github.com/apple/llvm-project/blob/3b34e59d28b1ce355850f89fb793c13bb6e0e469/clang/lib/Driver/ToolChain.cpp#L422).